themes/default/content/blog/build-images-50x-faster-docker-v4/index.md

```diff
@@ -0,0 +1,147 @@
+* BuildKit support (including cross-platform builds)
+* Rich Docker build logs inside Pulumi IaC
+* Pulumi YAML and Pulumi Java support
+
+<!--more-->
+
+### Significantly improved performance
+
+To improve the efficiency of the Docker Provider, we have changed the default behavior to rebuild only on context change. In addition, the `cacheFrom` option now offers the ability to use a registry as the image cache. By leveraging BuildKit and caching improvements, subsequent rebuilds may be up to 50x faster. This functionality and other improvements of the Docker Provider we are announcing today are [highly requested from our community](https://github.com/pulumi/pulumi-docker/issues/132).
+
+<!-- Add image/gif showing output during build, side by side comparison of old vs new?, or graph with speed comparison -->
+
+### BuildKit support
+
+The enhanced functionality and performance provided by the [Docker BuildKit](https://docs.docker.com/build/buildkit/) is now fully available to be leveraged in your Pulumi programs. BuildKit comes with functionality to improve your builds’ performance and the reusability of your Dockerfiles. Cross-platform builds are now supported, enabling image building for a target container runtime that is different from the build environment. BuildKit is now the default builder version mode. The V1 builder can still be accessed by specifying `Build.BuilderVersion=BuilderV1`.
+
+```typescript
+import * as docker from "@pulumi/docker";
+const img = new docker.Image("my-image", {
+    imageName: "docker.io/pulumibot/demo-image:latest",
+    build: {
+        context: "app",
+        dockerfile: "Dockerfile",
+        args: {
+            "BUILDKIT_INLINE_CACHE": "1"
+        },
+        builderVersion: "BuilderBuildKit", // can also be set to `BuilderV1`
+        cacheFrom: {
+            images: ["docker.io/pulumibot/demo-image:cache-base"]
+        },
+        platform: "linux/amd64",
+    }
+});
+```
+
+## Rich Docker build logs inside Pulumi IaC
+
+Filtered docker logs during build and push will now be displayed in the Info box on pulumi up.
+[![asciicast](https://asciinema.org/a/I8Xzmfme56ZP4uD6uo2U4i2wr.svg)](https://asciinema.org/a/I8Xzmfme56ZP4uD6uo2U4i2wr)
+
+### Pulumi YAML and Pulumi Java support
+
+We announced [Pulumi YAML and Pulumi Java](https://www.pulumi.com/blog/pulumi-universal-iac/) last year and are excited to have extended support to the Pulumi Docker Provider. This release schematizes all Docker resources, which enables auto-support of all Pulumi languages. In addition, this means the Docker image resource can now participate in the CRUD (create, replace, update, delete) lifecycle.
+
+### Migrating to 4.0
+
+The 4.0 version of the Docker provider consists of a significant overhaul of the Image Resource.
+These changes are mainly a result of either the way Pulumi generates language SDKs from a schematized resource, or of migrating to using the docker-go client library instead of invoking the Docker CLI.
```

Comment:
```suggestion
These changes are mainly a result of either the way Pulumi generates language SDKs from a schematized resource, or of using the docker-go client library instead of invoking the Docker CLI.
```